app/models/labware_creators/pcr_cycles_binned_plate/csv_file/row.rb

```diff
@@ -66,33 +83,45 @@ class PcrCyclesBinnedPlate::CsvFile::Row # rubocop:todo Metrics/ClassLength
              :submit_for_sequencing_column,
              :sub_pool_column,
              :coverage_column,
+             :hyb_panel_column,
              to: :header
 
-    # rubocop:todo Metrics/AbcSize,Metrics/CyclomaticComplexity,Metrics/MethodLength,Metrics/PerceivedComplexity
     def initialize(row_config, header, index, row_data)
       @row_config = row_config
       @header = header
       @index = index
       @row_data = row_data
 
-      # initialize supplied fields
+      initialize_sanger_supplied_fields
+      initialize_customer_supplied_fields
+
+      # initialize flag to indicate that sample should not be transferred
+      @do_not_transfer_sample = false
+    end
+
+    # rubocop:disable Metrics/AbcSize, Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
```

Comment:
Could these issues be resolved rather than suppressed?  What's been added doesn't look like it adds very much if any of these things.

Reply:
I often think rubocop is way too fussy. This class represents a row in a big spreadsheet, so there is a big list of fields to initialise when parsing the file. The field value extractions are not individually complex, there's just a lot of them. Splitting them out into groups as I've done is a bit clearer for the developer to see which values were pre-filled or filled by the customer (they download a csv, fill in some empty columns and re-upload it), but as you say doesn't make rubocop any happier.
If there's a better way to organise this I'd be happy to hear about it.

Reply:
I've had a quick look and can't identify anything easy.  I was hoping we could change the threshold of these checks for just this file but haven't been able to see how that can be done. You might be able to remove the supression of `Metrics/CyclomaticComplexity` since `Metrics/PerceivedComplexity` is a subclass of that.  Other than this, maybe we do just have to accept this.  The problem with doing that is that now it's a free for all for complexity between these disable/enable tags which might mean poor code in the future.